### DISCUSSION TITLE
Is there no synchronous version of Signer.signed_attrs() ?

### DISCUSSION THREAD

**Discussion by sbkis** [MatthiasValvekens/pyHanko#468] [Q&A]
Hey. I'm wondering if there isn't a synchronous version of `Signer.signed_attrs` and if there isn't how come this decision was made given that there are sync/async counterparts on most other methods. This one is async but doesn't have the `async`prefix on the name like other async methods and it looks like it was made async in 0.9 in a breaking change. Am I missing something here?

**Comment by MatthiasValvekens** [comment]
Any reason why you can't use `asyncio.run(...)`? Virtually all the synchronous methods in this lib are defined as compatibility wrappers for their async counterparts like that. If a synchronous implementation of `signed_attrs` were ever to be added, it'd be that kind of one-line wrapper. In other words, you might as well write it yourself, since the restrictions on `asyncio.run(...)` would apply either way.

As for the rationale: it's been several years (almost 3 years to the day, it seems!), and I honestly don't remember the details, but when I planned out the transition to async-first, I decided to provide compatibility wrappers for methods that I expected to be called within client code.
Not for everything, because it'd eventually result in a lot of never-removable code that presumably very few people would ever use--this was even more true 3 years ago than it was today. Also, the synchronous wrappers have not really been kept up-to-date API-wise anyway.
